ckanext/geodatagov/cli.py

```diff
@@ -425,9 +427,10 @@ def db_solr_sync(dryrun, cleanup_solr, update_solr):
 
     both = cleanup_solr == update_solr
     set_cleanup = {i if work_list[i] == "solr" else None for i in work_list} - {None}
```

Comment:
why `- {None}`?

Reply:
The set is generated from the set comprehension, in which there is potentially a None element.  We want to get rid of the None if it exists, therefore we use a set difference operator, `-`. This set  difference operation will collect all elements in the first set, which is `set_cleanup`,  but not in the second set, which is `{None}`.

Reply:
thanks! that's much clearer.